### ISSUE TITLE
IT locale formatRelative returns past days values without "last" word

### ISSUE THREAD

**Issue by davideas** [date-fns/date-fns#3488]
There is an issue with the [formatRelative function](https://date-fns.org/v2.30.0/docs/formatRelative) v2.30.0

  ## The problem

  ### 💻 Code demonstrating the problem

It happens _only_ when past day are Monday, Tuesday, Wednesday (not when is Sunday, Thursday etc..)
In Italian language some past days don't have the "Last" word displayed (see below the examples in Italian language).
In English all past days have the "Last" word correctly displayed.

```js
import { formatRelative } from 'date-fns';
import { it } from 'date-fns/locale';

// When formatLocale is Italian Locale object from date-fns.
formatLocale = it;
formatRelative(inputDate, new Date(), { locale: formatLocale, weekStartsOn: 1 });
  ```

  ### 🙁 Actual behavior
For example, if base date is _new Date()_ = 05/08/2023 18:29:26:

        inputDate value | formatRelative result
    02/08/2023 12:29:26 = mercoledì alle 12:29 (not ok)
    01/08/2023 18:29:26 = martedì alle 18:29 (not ok)
    31/07/2023 18:29:26 = lunedì alle 18:29 (not ok)
    30/07/2023 18:29:26 = domenica scorsa alle 18:29 (ok)
    29/07/2023 18:29:26 = sabato scorso alle 18:29 (ok)

  ### 🙂 Expected behavior
        inputDate value | formatRelative result
    02/08/2023 12:29:26 = mercoledì scorso alle 12:29
    01/08/2023 18:29:26 = martedì scorso alle 18:29
    31/07/2023 18:29:26 = lunedì scorso alle 18:29
    30/07/2023 18:29:26 = domenica scorsa alle 18:29
    29/07/2023 18:29:26 = sabato scorso alle 18:29

  ## Debug information

  - date-fns version: 2.30.0
  - Browser/Node.js version: does't matter
  - Your timezone: doesn't matter
  - Your current time: doesn't matter

**Comment by davideas** [comment]
When are you going to fix this tiny bug?

**Comment by doppiam** [comment]
Hi @davideas, I wanted to try fix this issue, I tried to replicate your error, but it seems to work fine.. am I missing something?

https://codesandbox.io/s/holy-bash-tn6l9k?file=/src/App.js

**Comment by davideas** [comment]
Yeah of course, because Last Thursday (giovedì scorso) is not impacted *only 3 days have the issue!* Please check my issue description.

**Comment by doppiam** [comment]
ok got it, now I understand the problem

https://codesandbox.io/s/holy-bash-tn6l9k?file=/src/App.js
this is updated and is showing the problem

**Comment by davideas** [comment]
@doppiam Waiting for a resolution version. Any date for new version?
